### PR TITLE
Duc/move fee account

### PR DESCRIPTION
**Pull request by ductm54**

### CODE REVIEW

File: contractsSol5/IKyberStorage.sol [hunk 1]
```diff
@@ -7,16 +7,17 @@ import "./IKyberNetworkProxy.sol";
...
         IKyberFeeHandler _feeHandler,
         address _matchingEngine
     ) external returns (bool);
 
     function setDAOContract(IKyberDAO _kyberDAO) external returns (bool);
 
-    function addReserve(address reserve, bytes32 reserveId)
-        external
-        returns (bool);
+    function addReserve(address reserve, bytes32 reserveId, ReserveType resType) external returns (bool);
```
Comment (ilanDoron) [inline contractsSol5/IKyberStorage.sol:20]: notice here code above is from pretify
So please use it on this file.

File: contractsSol5/KyberMatchingEngine.sol [hunk 2]
```diff
@@ -8,31 +8,23 @@ import "./IKyberStorage.sol";
...
- */
-contract KyberMatchingEngine is
-    KyberHintHandler,
-    IKyberMatchingEngine,
-    WithdrawableNoModifiers
-{
+*   @title Kyber matching engine contract
+*   The contract provides the following actions:
+*       - return list of reserves for a trade (getTradingReserves)
+*       - stores fee accounted data for reserve types
```
Comment (ilanDoron) [inline contractsSol5/KyberMatchingEngine.sol:14]: remove these two lines. not true any more

File: contractsSol5/mock/MaliciousMatchingEngine.sol [hunk 3]
```diff
@@ -8,13 +8,13 @@ contract MaliciousMatchingEngine is KyberMatchingEngine {
         KyberMatchingEngine(_admin)
     { /* empty body */ }
 
-    function getIsFeeAccountedReserves(bytes8[] memory reserveIds) internal view
-        returns(bool[] memory feeAccountedArr)
-    {
-        feeAccountedArr = new bool[](reserveIds.length);
+    // function getIsFeeAccountedReserves(bytes8[] memory reserveIds) internal view
```
Comment (ilanDoron) [inline contractsSol5/mock/MaliciousMatchingEngine.sol:11]: why not delete?

File: contractsSol5/mock/OtherMatchingEngine.sol [hunk 4]
```diff
@@ -9,20 +9,20 @@ contract OtherMatchingEngine is KyberMatchingEngine {
...
+    //     reserveType[reserveId] = uint(resType);
+    //     return false;
+    // }
 
-    function removeReserve(bytes32 reserveId) external returns (bool) {
-        onlyNetwork();
-        reserveType[reserveId] = uint(ReserveType.NONE);
-        return false;
-    }
+    // function removeReserve(bytes32 reserveId) external returns (bool) {
```
Comment (ilanDoron) [inline contractsSol5/mock/OtherMatchingEngine.sol:21]: why not delete all commented out?

File: test/sol5/kyberMatchingEngine.js [hunk 5]
```diff
@@ -113,18 +114,18 @@ contract('KyberMatchingEngine', function(accounts) {
...
         it("should not have unauthorized personnel set fee accounted data", async() => {
             await expectRevert(
-                matchingEngine.setFeeAccountedPerReserveType(true, true, true, false, true, true, {from: operator}),
+                storage.setFeeAccountedPerReserveType(true, true, true, false, true, true, {from: operator}),
                 "only admin"
             );
 
             await expectRevert(
-                matchingEngine.setFeeAccountedPerReserveType(true, true, true, false, true, true, {from: network}),
+                storage.setFeeAccountedPerReserveType(true, true, true, false, true, true, {from: network}),
```
Comment (ilanDoron) [inline test/sol5/kyberMatchingEngine.js:122]: please don't test this in matchingEngine test file.

move to kyberStorage test file.

File: contractsSol5/IKyberStorage.sol (hunk 1 above)
Comment (ductm54) [reply contractsSol5/IKyberStorage.sol]: udpated

File: contractsSol5/KyberMatchingEngine.sol (hunk 2 above)
Comment (ductm54) [reply contractsSol5/KyberMatchingEngine.sol]: removed

File: contractsSol5/mock/MaliciousMatchingEngine.sol (hunk 3 above)
Comment (ductm54) [reply contractsSol5/mock/MaliciousMatchingEngine.sol:11]: we might need to read those when it come to coverage test

File: test/sol5/kyberMatchingEngine.js (hunk 5 above)
Comment (ductm54) [reply test/sol5/kyberMatchingEngine.js]: moved

**Pull request merged**